pyFV3/stencils/xppm.py

```diff
@@ -310,21 +341,20 @@ def __init__(
         # Arguments come from:
         # namelist.grid_type
         # grid.dxa
-        if grid_type == 3 or grid_type > 4:
-            raise NotImplementedError(
-                "X Piecewise Parabolic (xppm): "
-                f" grid type {grid_type} not implemented. <3 or 4 available."
-            )
-
-        if abs(iord) >= 8 and iord != 8:
+        available_grid_options = [0, 4]
+        if grid_type not in available_grid_options:
             raise NotImplementedError(
-                "X Piecewise Parabolic (xppm): "
-                f"iord {iord} != 8 not implemented when >= 8."
+                "Y Piecewise Parabolic (yppm) configuration: "
+                f"grid type {grid_type} not implemented. "
+                f"Options are {available_grid_options}."
             )
 
-        if iord < 0:
+        available_iords = [-6, 5, 6, 8]
```

Comment:
You say iord must be positive (line 329), but have a value of -6 here?

Reply:
Yes the `docs` are not corrected there. I also need to have XPPM docs point here, as they share a same conceptual limitation.

Reply:
It would be great if we could use something more descriptive than integers here, but I understand that might not be possible.

Reply:
Allowing `Enum` to be parsed should be do-able. We just need to boil down the enum to an AST unique and then either we can convert it silently to an integer in code for portability